pkg/sidecar/config.go

```diff
@@ -114,16 +120,7 @@ func ValidateConfig(c *Config) error {
 		c.RenewSignal = c.RenewSignalDeprecated
 	}
 
-	switch {
-	case c.SvidFileName == "":
-		return errors.New("svid_file_name is required")
-	case c.SvidKeyFileName == "":
-		return errors.New("svid_key_file_name is required")
-	case c.SvidBundleFileName == "":
-		return errors.New("svid_bundle_file_name is required")
-	default:
-		return nil
-	}
+	return nil
```

Comment:
Should we do
```golang
if c.SvidFileName == "" && c.JWTSvidFilename == "" {
    return errors.New("svid_file_name and/or jwt_svid_file_name is required")
}
if c.SvidFileName != "" && c.SvidKeyFileName == "" {
    return errors.New("svid_key_file_name is required when using svid_file_name")
}
if c.SvidFileName != "" && c.SvidBundleFileName == "" {
    return errors.New("svid_bundle_file_name is required when using svid_file_name")
}
if c.JWTFilename != "" && c.JWTBundleFilename == "" {
    return errors.New("jwt_bundle_file_name is required when using jwt_svid_file_name")
}
```
?

Reply:
What do you think about this:

```golang
if c.SvidFileName == "" && c.JWTSvidFilename == "" && c.JWTBundleFilename == "" {
		return errors.New("svid_file_name, jwt_svid_file_name or jwt_bundle_file_name is required")
	}

	if c.SvidFileName != "" && c.SvidKeyFileName == "" {
		return errors.New("svid_key_file_name is required when using svid_file_name")
	}

	if c.SvidFileName != "" && c.SvidBundleFileName == "" {
		return errors.New("svid_bundle_file_name is required when using svid_file_name")
	}

	if c.JWTSvidFilename != "" && c.JWTAudience == "" {
		return errors.New("jwt_audience is required when using jwt_svid_file_name")
	}
}
```

I think there are three cases:
* Fetch X.509 certificates: SvidKeyFileName and SvidBundleFIleName should be filled.
* Fetch JWT SVID: JWTAudience should be filled.
* Fetch JWT bundle.

Reply:
@keeganwitt, @JU4N98  Correct me if I'm wrong, we have two sets of properties:

1.  `c.SvidFileName`, `c.SvidKeyFilename`, `c.SvidBundleFileName`
2.  `c.JWTFilename`, `c.JWTBundleFilename`, `c.JWTAudience`

I understand that we need to validate that in each group the properties are all or none present, and also at least one group is present.

We could do it with something like this (please review before applying this changes):

```
// Count the number of non-empty properties for the first set
firstGroupCount := countNonEmpty(c.SvidFileName, c.SvidKeyFilename, c.SvidBundleFileName)

// Check if either all or none are set for the first set
if firstGroupCount != 0 && firstGroupCount != 3 {
    return errors.New("all or none of 'svid_file_name', 'svid_key_file_name', 'svid_bundle_file_name' must be specified")
}

// Same for the other set
secondGroupCount := countNonEmpty(c.JWTFilename, c.JWTBundleFilename, c.JWTAudience)
if secondGroupCount != 0 && secondGroupCount != 3 {
    return errors.New("all or none of 'jwt_filename', 'jwt_bundle_file_name', 'jwt_audience' must be specified")
}

// Ensure that at least one of the groups is fully set
if firstGroupCount != 3 && secondGroupCount != 3 {
    return errors.New("at least one of the sets ('svid_file_name', 'svid_key_file_name', 'svid_bundle_file_name') or ('jwt_filename', 'jwt_bundle_file_name', 'jwt_audience') must be fully specified")
}

func countNonEmpty(strings ...string) int {
    count := 0
    for _, str := range strings {
        if str != "" {
            count++
        }
    }
    return count
}
```

Reply:
Although on a different approach, we could consider scenarios where a workload might only need the X.509 bundles or the JWT token, but not the rest. If we were to enable such flexibility, we could use the following logic:

1. X.509 Properties:
    - If `svidFileName` is provided, `svidKeyFilename` must also be provided.
    - `svidBundleFileName` is optional.

2. JWT Properties:
    - If `jwtFileName` is provided, `jwtAudience` must also be provided.
    - `jwtBundleFilename` is optional.

The crucial rule here is that at least one of the four elements (`svidFileName`, `svidKeyFilename`, `jwtFileName`, `jwtAudience`) must be present. This ensures that there is at least a basic configuration for the workload to operate with, while still providing the flexibility for more specific use cases.

I believe this approach could provide flexibility for various use cases, but I'll leave it to the maintainers @MarcosDY and @faisal-memon to decide the best course of action.

Reply:
I'd recommend aligning JWT settings with X.509 config for consistency. Consult with maintainers for final decision.

Reply:
Right now we require all three x509 properties to be set. I think its best to continue that approach, so require all three JWT properties to be set or none. @JU4N98 you can follow the approach @maxlambrecht laid out in this comment: https://github.com/spiffe/spiffe-helper/pull/85#discussion_r1394791136

If someone has a use case that requires more flexibility we can address that in a separate pr.

Reply:
Awesome, I just upload the last changes.

Reply:
uffff too many configurations add a lot of complexity to code and users... 

I'm starting to think that we must have a single folder for X.509 SVID... defaulting names...
and same for JWT... and in JWT it is possible to a user does not want to have bundles.... but only Token (because they can depends on OIDC to get bundles)
Any thoughts about that?

Reply:
@MarcosDY that all sounds good to me, but out of scope for this pr. Let's merge this one and then make those changes after.

Reply:
@faisal-memon @maxlambrecht a related question to this, is that we allow to set token and bundle but not only one of them, I have the feeling that it is separated, since jwt are different to x509, 
I can think in cases where a token is required but they don't care about bundles, so I'm not sure if it worth to make all required.
What is the reason to keep this way? is it only for consistency?

Reply:
Thanks @JU4N98 for updating the PR. @MarcosDY you can now specify to just download the bundle or the token.